examples/platform/silabs/SilabsDeviceAttestationCreds.cpp

```diff
@@ -33,12 +37,26 @@ namespace {
 
 class DeviceAttestationCredsSilabs : public DeviceAttestationCredentialsProvider
 {
+    static constexpr size_t kDebugLength = 8;
 
 public:
-    CHIP_ERROR GetCertificationDeclaration(MutableByteSpan & out_buffer) override
+    CHIP_ERROR GetCertificationDeclaration(MutableByteSpan & out_span) override
     {
-        ByteSpan cd_span(((uint8_t *) __attestation_credentials_base) + SILABS_CREDENTIALS_CD_OFFSET, SILABS_CREDENTIALS_CD_SIZE);
-        return CopySpanToMutableSpan(cd_span, out_buffer);
+        uint32_t offset = SILABS_CREDENTIALS_CD_OFFSET;
+        uint32_t size = SILABS_CREDENTIALS_CD_SIZE;
+
+        if(SilabsConfig::ConfigValueExists(SilabsConfig::kConfigKey_Creds_CD_Offset) &&
+           SilabsConfig::ConfigValueExists(SilabsConfig::kConfigKey_Creds_CD_Size))
+        {
+            ReturnErrorOnFailure(SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_CD_Offset, offset));
+            ReturnErrorOnFailure(SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_CD_Size, size));
+        }
+
+        uint8_t * address = _credentials_address + offset;
+        ByteSpan cd_span(address, size);
+        ChipLogProgress(DeviceLayer, "GetCertificationDeclaration, addr:%p, size:%lu", address, size);
+        ChipLogByteSpan(Zcl, ByteSpan(cd_span.data(), kDebugLength));
```

Comment:
Are we guaranteed that `kDebugLength < size` here?  Might be worth documenting why, if we are.  Same in the other places like this below.

Reply:
Miss-aligned certificates is a common error, and printing the first few bytes is quite useful to validate this. Eight is just enough for this purpose. I will add this comment into the code.

Reply:
My point is that if `kDebugLength > size` then we will be printing random data.  But presumably `size` is bigger than `kDebugLength` in practice.

Reply:
Size is less than 8 only if the files are missing, or corrupted. I have added a check to cover this cases.